src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1525,1898 @@ interface MPSMatrixMultiplication {
+	[BaseType (typeof(MPSCnnConvolutionDescriptor), Name = "MPSCNNSubPixelConvolutionDescriptor")]
+	interface MPSCnnSubPixelConvolutionDescriptor
+	{
+		[Export ("subPixelScaleFactor")]
+		nuint SubPixelScaleFactor { get; set; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnConvolutionDescriptor), Name = "MPSCNNDepthWiseConvolutionDescriptor")]
+	interface MPSCnnDepthWiseConvolutionDescriptor
+	{
+		[Export ("channelMultiplier")]
+		nuint ChannelMultiplier { get; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSState), Name = "MPSCNNConvolutionState")]
+	interface MPSCnnConvolutionState : MPSImageSizeEncodingState
+	{
+		[Export ("kernelWidth")]
+		nuint KernelWidth { get; }
+
+		[Export ("kernelHeight")]
+		nuint KernelHeight { get; }
+
+		[Export ("sourceOffset")]
+		MPSOffset SourceOffset { get; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnKernel), Name = "MPSCNNConvolutionTranspose")]
+	interface MPSCnnConvolutionTranspose
+	{
+		[Export ("inputFeatureChannels")]
+		nuint InputFeatureChannels { get; }
+
+		[Export ("outputFeatureChannels")]
+		nuint OutputFeatureChannels { get; }
+
+		[Export ("kernelOffsetX")]
+		nint KernelOffsetX { get; set; }
+
+		[Export ("kernelOffsetY")]
+		nint KernelOffsetY { get; set; }
+
+		[Export ("groups")]
+		nuint Groups { get; }
+
+		[Export ("initWithDevice:weights:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, IMPSCnnConvolutionDataSource weights);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+
+		[Export ("encodeToCommandBuffer:sourceImage:convolutionState:")]
+		MPSImage Encode (IMTLCommandBuffer commandBuffer, MPSImage sourceImage, [NullAllowed]MPSCnnConvolutionState convolutionState);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnKernel), Name = "MPSCNNBinaryConvolution")]
+	interface MPSCnnBinaryConvolution
+	{
+		[Export ("inputFeatureChannels")]
+		nuint InputFeatureChannels { get; }
+
+		[Export ("outputFeatureChannels")]
+		nuint OutputFeatureChannels { get; }
+
+		[Export ("initWithDevice:convolutionData:scaleValue:type:flags:")]
+		IntPtr Constructor (IMTLDevice device, IMPSCnnConvolutionDataSource convolutionData, float scaleValue,MPSCnnBinaryConvolutionType type,MPSCnnBinaryConvolutionFlags flags);
+
+		[Export ("initWithDevice:convolutionData:outputBiasTerms:outputScaleTerms:inputBiasTerms:inputScaleTerms:type:flags:")]
+		IntPtr Constructor (IMTLDevice device, IMPSCnnConvolutionDataSource convolutionData, [NullAllowed] IntPtr /* float* */ outputBiasTerms, [NullAllowed] IntPtr /* float* */ outputScaleTerms, [NullAllowed] IntPtr /* float* */ inputBiasTerms, [NullAllowed] IntPtr /* float* */ inputScaleTerms,MPSCnnBinaryConvolutionType type, MPSCnnBinaryConvolutionFlags flags);
```

Comment:
I think the code-behind implementation is sufficient, so [Internal] to hide the IntPtr signature.